src/analyses/accessAnalysis.ml

```diff
@@ -0,0 +1,231 @@
+(** Access and data race analysis. *)
+
+module M = Messages
+module LF = LibraryFunctions
+open Prelude.Ana
+open Analyses
+open GobConfig
+
+
+(** Access and data race analyzer without base --- this is the new standard *)
+module Spec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "access"
+
+  (** Add current lockset alongside to the base analysis domain. Global data is collected using dirty side-effecting. *)
+  module D = Lattice.Unit
+  module C = Lattice.Unit
+
+  module G =
+  struct
+    include Access.PM
+
+    let leq x y = !GU.postsolving || leq x y (* HACK: to pass verify*)
+  end
+  module V = Printable.Prod (Access.LVOpt) (Access.T)
+
+  let safe       = ref 0
+  let vulnerable = ref 0
+  let unsafe     = ref 0
+
+  let init marshal =
+    safe := 0;
+    vulnerable := 0;
+    unsafe := 0
+
+  let side_access ctx ty lv_opt ls_opt (conf, w, loc, e, lp) =
+    if !GU.should_warn then (
+      let d =
+        let open Access in
+        PM.singleton ls_opt (
+          AS.singleton (conf, w, loc, e, lp)
+        )
+      in
+      ctx.sideg (lv_opt, ty) d
+    )
+    else
+      ctx.sideg (lv_opt, ty) (G.bot ()) (* HACK: just to pass validation with MCP DomVariantLattice *)
```

Comment:
Why is this needed? would these not the default value `bot` anyway?

Reply:
The problem is that they're different bottoms. If nothing is side-effected to these variables during the actual solve, then they have the bottom from the `Lift2` of local and global domains, not the bottom of `DomVariantLattice` or this `G`.

This forces all the lifted layers to be created such that during verify the hacky `G.leq` from above is actually called. Otherwise we'd just be calling the `leq` for the outermost `Lift2`, which we don't want to allow such verification bypass in and everything would fail verification.

This is the unfortunate side effect of only wanting to collect accesses into the global invariant during postsolving, but still pass the verification, which happens during the same postsolving.